plugins/azure/resoto_plugin_azure/resource/base.py

```diff
@@ -96,7 +96,7 @@ def delete(self, graph: Graph) -> bool:
         Returns:
         bool: True if the resource was successfully deleted; False otherwise.
         """
-        subscription_id = self.resource_subscription_id()
+        subscription_id = self.resource_subscription_id() or ""
```

Comment:
No. this or statement does not make any sense. Please also fix all other `or ""` occurrences.